dev_mode/webpack.config.js

```diff
@@ -160,49 +160,68 @@ function ignored(path) {
 // Set up module federation sharing config
 const shared = {};
 
+// eager so that built-in extensions can be bundled together into just a few
+// js files to load
+const eagerPackages = new Set(Object.keys(package_data.resolutions));
+for (let pkg of Object.keys(jlab.linkedPackages)) {
+  eagerPackages.delete(pkg);
+}
+
 // Make sure any resolutions are shared
-for (let [key, requiredVersion] of Object.entries(package_data.resolutions)) {
-  // eager so that built-in extensions can be bundled together into just a few
-  // js files to load
-  shared[key] = { requiredVersion, eager: true };
+for (let [pkg, requiredVersion] of Object.entries(package_data.resolutions)) {
+  shared[pkg] = { requiredVersion };
+  if (eagerPackages.has(pkg)) {
+    shared[pkg].eager = true;
+  }
 }
 
 // Add any extension packages that are not in resolutions (i.e., installed from npm)
 for (let pkg of extensionPackages) {
-  if (shared[pkg] === undefined) {
+  if (!shared[pkg]) {
     shared[pkg] = {
-      requiredVersion: require(`${pkg}/package.json`).version,
-      eager: true
+      requiredVersion: require(`${pkg}/package.json`).version
     };
+    if (eagerPackages.has(pkg)) {
+      shared[pkg].eager = true;
+    }
   }
 }
 
 // Add dependencies and sharedPackage config from extension packages if they
 // are not already in the shared config. This means that if there is a
 // conflict, the resolutions package version is the one that is shared.
 extraShared = [];
```

Comment:
Speaking of linting, looks like this var is missing a const:

```suggestion
const extraShared = [];
```

Reply:
Weird that lint didn't pick that up.

Reply:
(or maybe it did as a warning, one of the hundreds of warnings that would be good to sprint on)